streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -552,6 +553,32 @@ public Subscription subscription(Set<String> topics) {
         return assignment;
     }
 
+    // visible for testing
+    List<TaskId> interleaveTasksByGroupId(Collection<TaskId> taskIds) {
```

Comment:
Note that in line 523 above, we are jumping by num. consumers round-robin to assign tasks. With this interleaving tasks list, the jumping of num. consumers does not make much more sense right?

Reply:
No, it doesn't, but I still have to come up with a better solution and wanted eyes on the interleaving strategy, maybe I should change the title to `WIP`?